llarp/router/router.cpp

```diff
@@ -737,6 +737,14 @@ namespace llarp
         return !_rcLookupHandler.RemoteIsAllowed(rc.pubkey);
       });
     }
+    else
+    {
+      nodedb()->RemoveIf([&](const RouterContact &rc) -> bool {
+        if(IsBootstrapNode(rc.pubkey))
+          return false;
+        return not rc.IsPublicRouter();
+      });
+    }
 
```

Comment:
This logic will run if we haven't received the whitelist yet. Will that cause everything to be removed in this case? That would essentially mean we wipe our nodedb at startup.

Reply:
good point, adding check.